src/lib/commandclass/ColorSwitchCC.ts

```diff
@@ -344,68 +375,65 @@ export class ColorSwitchCCSupportedReport extends ColorSwitchCC
 			supportsIndex: Boolean(this.payload[1] & 1),
 		};
 
-		const node = this.getNode()!;
-		const endpoint = this.getEndpoint()!;
-
+		const valueDB = this.getValueDB();
 		for (const key of ColorKeys) {
 			const supportsColorKey = supportedColorTableKey(key);
 
-			const valueId: ValueID = {
-				commandClass: CommandClasses["Color Switch"],
-				endpoint: endpoint.index,
-				property: "supportedColor",
-				propertyKey: key,
-			};
+			const valueId: ValueID = getSupportedColorValueID(
+				this.endpointIndex,
+				supportsColorKey,
+			);
 
-			// TODO: Should we be setting this every run?
-			node.valueDB.setMetadata(valueId, {
-				...ValueMetadata.ReadOnly,
-				label: `Supports ${key}`,
-				description: `Whether the endpoint supports setting the ${key} color.`,
-			});
-			node.valueDB.setValue(
+			if (!valueDB.hasMetadata(valueId)) {
+				valueDB.setMetadata(valueId, {
+					...ValueMetadata.ReadOnly,
+					label: `Supports ${key}`,
+					description: `Whether the endpoint supports setting the ${key} color.`,
+				});
+			}
+			valueDB.setValue(
 				valueId,
 				this._supportedColors[supportsColorKey] ?? false,
 			);
 		}
```

Comment:
I guess this makes sense here - I've usually done this kind of stuff in the interview method instead to keep the CC constructors shorter.

Reply:
I moved metadata initialization to interview.

It is being done only if the metadata does not already exist, and for a complete interview.  Should it always be done regardless of complete / exists, in case we change the metadata in the future?